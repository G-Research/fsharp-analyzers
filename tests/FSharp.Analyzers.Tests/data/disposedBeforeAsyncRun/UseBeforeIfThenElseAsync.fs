module M

open System

let f1() =
    use disp = new IO.FileStream("", IO.FileMode.Append)
    if true then
        async {
            return ""
        }
    else 
        failwith ""

let f2() =
    use disp = new IO.FileStream("", IO.FileMode.Append)
    if true then
        failwith ""
    else
        async {
            return ""
        } 
