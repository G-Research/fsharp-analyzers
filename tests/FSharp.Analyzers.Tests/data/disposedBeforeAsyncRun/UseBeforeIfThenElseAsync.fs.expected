GRA-DISPBEFOREASYNC-001 | Warning | (6,8 - 6,12) | Object is disposed before returned async is run | []
GRA-DISPBEFOREASYNC-001 | Warning | (15,8 - 15,12) | Object is disposed before returned async is run | []
